src/zenml/new/pipelines/pipeline.py

```diff
@@ -428,11 +431,50 @@ def prepare(self, *args: Any, **kwargs: Any) -> None:
         Args:
             *args: Pipeline entrypoint input arguments.
             **kwargs: Pipeline entrypoint input keyword arguments.
+
+        Raises:
+            RuntimeError: If the pipeline has parameters configured differently in
+                configuration file and code.
         """
         # Clear existing parameters and invocations
         self._parameters = {}
         self._invocations = {}
 
+        conflicting_parameters = {}
+        parameters_ = (self.configuration.parameters or {}).copy()
+        if from_file_ := self._from_config_file.get("parameters", None):
+            parameters_ = dict_utils.recursive_update(parameters_, from_file_)
+        if parameters_:
+            for k, v_runtime in kwargs.items():
+                if v_config := parameters_.get(k, None):
```

Comment:
`None` or `False` might also be valid parameter values which will not get detected by this logic I think?